charmcraft/commands/store/__init__.py

```diff
@@ -461,3 +467,109 @@ def run(self, parsed_args):
         table = tabulate(data, headers=headers, tablefmt='plain', numalign='left')
         for line in table.splitlines():
             logger.info(line)
+
+
+def _get_lib_info(*, full_name=None, lib_path=None):
+    """Get the whole lib info from the path/file."""
+    if full_name is None:
+        # get it from the lib_path
+        bad_structure_msg = (
```

Comment:
wouldn't it be better to have a couple of `BadLib{Path,Name}(CommandError)` classes, rather than building these strings on the fast path?

Reply:
Great idea! Did that now.